app/src/ui/create-branch/create-branch-dialog.tsx

```diff
@@ -392,13 +401,27 @@ export class CreateBranch extends React.Component<
     )
 
     // there are no relevant rules for this branch name, so return
-    if (toCheckForBypass.size === 0) {
+    if (toCheck.size === 0) {
+      return
+    }
+
+    // check for actual failures
+    const { branchNamePatterns, creationRestricted } = await parseRepoRules(
+      branchRules,
+      this.props.cachedRepoRulesets,
```

Comment:
So, I ran into an issue with using this `cachedRepoRulesets`. When I released to beta and tested on Windows, I opened `desktop/desktop` in GitHub Desktop. Then, opened `desktop/desktop` in dotcom and added a branch name pattern rule. Then, I returned to GitHub Desktop and attempted to create a branch that violated that rule. It did not catch even tho the branch rule was being pulled down here because these cached rules were empty. I had to switch repositories to get the `app-store.ts` to rule to populate this cache.

Two thoughts: 
1. We could use whatever we pull down and not pass in the cache here.
2. We could do something along the lines that Markus had pointed out in one of the other repo rule fix PRs and make the `app-store.ts` logic run on window focus.

Reply:
(Note this is an edge case/only when fiddling with the repo rules while desktop is open)

Reply:
The endpoint queried by this component is ["Get rules for a branch"](https://docs.github.com/en/rest/repos/rules?apiVersion=2022-11-28#get-rules-for-a-branch), which only returns a very slim list of rules. The user's bypass status comes from the full ruleset object, which is why these cached rulesets are passed in at all. Option 1 therefore wouldn't really work here.

Option 2 would work, but users could potentially end up with a lot of queries. The ["Get all repository rulesets" endpoint](https://docs.github.com/en/rest/repos/rules?apiVersion=2022-11-28#get-all-repository-rulesets) doesn't return this bypass status either, so each individual ruleset has to be queried, and a given repo could have any number of rulesets. That's why I opted for heavier caching like this: These rulesets are probably rarely changing in the first place outside of testing, and swapping it to something like the window focus event would likely trigger a _lot_ more API calls than are necessary.

Reply:
> so each individual ruleset has to be queried, and a given repo could have any number of ruleset

Wonder if it would be possible for client app consumption to add an api call that simply returns the last updated date of any rules? That we do that one more frequently and only go get all the rules if there has been an update.